### PR TITLE
Create book options

### PR DESCRIPTION
**Pull request by iamsjunaid**
Pull Request Summary:

**Title**: Adding Options for Listing Books and Labels, and Book Creation

**Description**:
This pull request introduces new features to enhance the functionality of the catalog system. The main focus is on providing the following options:

1. **List All Books**: Users can now easily view a list of all available books in the catalog.

2. **List All Labels**: The system now supports listing various labels, such as 'Gift' and 'New', which helps users identify different types of items.

3. **Add a Book**: A new method has been implemented in the `app.rb` file to allow users to add new books to the catalog.

**Changes Made**:

1. Modified `book.rb`: The `book.rb` file has been updated to include a new `to_s` method. This method allows the data of each book to be presented in a user-friendly format when printing.

2. Modified `label.rb`: Similar to `book.rb`, the `label.rb` file has been modified to include a `to_s` method that formats label data for better readability.

3. Created `app.rb`: A new method named `add_book` has been added to the `app.rb` file. This method prompts users for necessary parameters and creates a new book instance. Additionally, minor improvements have been made to ensure a smoother user experience.

**Testing**:
- The new features have been thoroughly tested to ensure proper functionality.

**Notes**:
This pull request aims to enhance the user experience by offering convenient options for viewing and managing books and labels within the catalog system. The code changes have been well-tested and maintain the overall structure of the project. Your feedback is highly appreciated!

### CODE REVIEW

File: classes/label.rb [hunk 1]
```diff
@@ -4,6 +4,7 @@ class Label
 
   def initialize(params)
     @id = params[:id] || Random.rand(1..1000)
+    @name = params[:name]
```
Comment (luigirazum) [inline classes/label.rb:7]: Hello @iamsjunaid,
- Quick question... Where did 'name' come from?

I would like to suggest you kindly to remove it...

_According to the UML diagram_
image

File: classes/label.rb [hunk 2]
```diff
@@ -13,4 +14,17 @@ def add_item(item)
     @items << item unless @items.include?(item)
     item.add_label(self)
   end
+
+  private
+
+  def to_s
+    properties = [
+      "id: #{@id}",
+      "name: #{@name}",
```
Comment (luigirazum) [inline classes/label.rb:23]: As in the 'constructor' method... where did 'name' come from?

- I would suggest you kindly to remove it.

File: app/app.rb [hunk 3]
```diff
@@ -107,6 +112,8 @@ def create_item(type, params)
     case type
     when :album
       MusicAlbum.new(params)
+    when :book
+      Book.new(params)
```
Comment (luigirazum) [inline app/app.rb:116]: ## 👏 Well done!!
We can easily another type of collections to our app.

File: app/app.rb [hunk 4]
```diff
@@ -1,17 +1,21 @@
...
 class App
-  attr_reader :genres, :albums
+  attr_reader :genres, :albums, :books, :labels
 
   def initialize(params = { data_dir: 'data' })
     @data_dir = params[:data_dir]
     @genres = params[:genres] || load(:genres)
     @albums = params[:albums] || load(:albums)
+    @labels = params[:labels] || load(:labels)
+    @books = params[:books] || load(:books)
```
Comment (luigirazum) [inline app/app.rb:14]: ## 😮 Just like this!
And our app can use books/labels.

**Review by luigirazum — CHANGES_REQUESTED** [review]
Hi @iamsjunaid, I'm Luis your code reviewer

<br>Good job so far! <br>👏👏👏 
---
Status: ♻️ Changes Required
===


There are some issues that you still need to work on to go to the next project, but you're almost there!
## To Highlight 🎉
- No Linters errors ✔️
- Clean Code 👏 ✔️
- Outstanding PR description ✔️

 ## Required Changes ♻️
- [ ] The UML diagram doesn't show any 'name' property for 'Label' class.
- [ ] **_Please 👀 Check the comments below_**

## Optional suggestions ✍️
- N/A
_Every comment with the <kbd>OPTIONAL</kbd> prefix is not crucial enough to stop the approval of this PR. <br>However, I strongly recommend you take them into account as they can make your code better._

***
Cheers and Happy Coding! 👨‍💻
---
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me `@luigirazum` in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. <br>You should use the same Pull Request submitted for the first review,<br>either valid or invalid unless it is requested otherwise._** 

----
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/label.rb (hunk 1 above)
Comment (iamsjunaid) [reply classes/label.rb:7]: that was a typo😁!!! I forgot to delete it.

**Review by luigirazum — APPROVED** [review]
STATUS: 🟢 APPROVED!!!  🎉 🎉 🎉
===
Hi @iamsjunaid, 🙋‍♂️ I'm Luis your code reviewer. 
---
### Great job !!! 👏 👏 👏 
Your project is **complete!**
Good work on making the changes requested in the previous review.

Now you can merge this branch and move on 🥇 
---
Congratulations! :champagne: :champagne: :champagne:
===
## To Highlight 🎉
- No Linters errors ✔️
- Good commit messages.



## Optional suggestions ✍️
- N/A
> _Every comment with the <kbd>OPTIONAL</kbd> prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

***
Happy Coding! 👨‍💻 and Good Luck 🍀 with the coming ones 🤞
---
Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me `@luigirazum` in your question so I can receive the notification, you can also reach me out on Slack using the same tag as well.

> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._** 

----
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**